src/NuGet.Core/Microsoft.Build.NuGetSdkResolver/NuGetSdkResolver.cs

```diff
@@ -130,15 +131,16 @@ public static SdkResult GetSdkResult(SdkReference sdk, object nuGetVersion, SdkR
                         DefaultCredentialServiceUtility.SetupDefaultCredentialService(nugetSDKLogger, nonInteractive: true);
 
                         // Asynchronously run the restore without a commit which find the package on configured feeds, download, and unzip it without generating any other files
-                        var results = RestoreRunnerEx.RunWithoutCommit(
-                                context.ProjectFilePath,
-                                sdk.Name,
-                                parsedSdkVersion.ToFullString(),
-                                settings,
-                                nugetSDKLogger)
-                            .ConfigureAwait(continueOnCapturedContext: false)
-                            .GetAwaiter()
-                            .GetResult();
+                        // This must be run in its own task because legacy project system evaluates projects on the UI thread which can cause RunWithoutCommit() to deadlock
+                        // https://developercommunity.visualstudio.com/content/problem/311379/solution-load-never-completes-when-project-contain.html
+                        var restoreTask = Task.Run(() => RestoreRunnerEx.RunWithoutCommit(
```

Comment:
As I recall, there are some rules we're supposed to follow about using the thread pool in VS. Unfortunately, I don't know them and internalized "don't do it". @cdmihai may have a better understanding?

Reply:
Have you repro's it locally and made sure this fixes the issue?

Reply:
I have repro'd it locally and verified that this fixed it.